<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Signature Regulatory Clustering • SiRCleR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="apple-touch-icon-60x60.png">
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><link href="extra.css" rel="stylesheet">
<meta property="og:title" content="Signature Regulatory Clustering">
<meta name="description" content="The SiRCle method integrates DNA methylation, RNA-seq and proteomics data at the gene level to deconvolute the association between dysregulation within and across possible regulatory layers (DNA methylation, transcription and/or translation).Based on logical regulatory rules, `sircleRCM`, genes are grouped in SiRCle clusters based on the layer (DNA methylation, transcription and/or translation) where dys-regulation first occurs. Using the SiRCle clusters, one can find the primary biological processes altered by applying Over Representation Analysis (ORA) and the drivers behind it using Transcription Factor (TF) analysis.">
<meta property="og:description" content="The SiRCle method integrates DNA methylation, RNA-seq and proteomics data at the gene level to deconvolute the association between dysregulation within and across possible regulatory layers (DNA methylation, transcription and/or translation).Based on logical regulatory rules, `sircleRCM`, genes are grouped in SiRCle clusters based on the layer (DNA methylation, transcription and/or translation) where dys-regulation first occurs. Using the SiRCle clusters, one can find the primary biological processes altered by applying Over Representation Analysis (ORA) and the drivers behind it using Transcription Factor (TF) analysis.">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">SiRCleR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-tutorials" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true"><span class="fa fa-book-open"></span> Tutorials</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-tutorials">
<li><a class="dropdown-item" href="articles/ChooseSettings.html">ChooseSettings</a></li>
    <li><a class="dropdown-item" href="articles/DataAnalysisExample.html">TutorialR</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="reference/index.html"><span class="fa fa-solid fa-rocket"></span> Reference</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/ArianeMora/SiRCleR"><span class="fa fa-brands fa-github"></span></a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/ArianeMora/scircm"><span class="fa fa-brands fa-python"></span></a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://genomemedicine.biomedcentral.com/articles/10.1186/s13073-024-01415-3"><span class="fa fa-solid fa-dna"></span> Manuscript</a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9">
<div class="section level1">
<div class="page-header"><h1 id="short-introduction">Short Introduction<a class="anchor" aria-label="anchor" href="#short-introduction"></a>
</h1></div>
<p>The <strong>S</strong>ignature <strong>R</strong>egulatory <strong>Cl</strong>ust<strong>e</strong>ring (<strong><em>SiRCle</em></strong>) method has first been released as a biorxiv preprint and since then we have enhanced the method further as part of our revision in genome medicine.<br>
The <strong><em>SiRCle</em></strong> method was first developed in 2022 to integrate DNA methylation, RNA-seq and proteomics data at the gene level to deconvolute the association between dysregulation within and across possible regulatory layers (DNA methylation, transcription and/or translation). The results of the study were published on biorxiv in a shared effort by <a href="https://www.biorxiv.org/content/10.1101/2022.07.02.498058v1" class="external-link">Mora &amp; Schmidt et. al.</a>. As part of the revisions in Genome Medicine, we have updated some parts of the SiRCle clustering method and extended our methods section in the manuscript. We also added new data and a PanCan analysis. All of this can be found in <a href="https://genomemedicine.biomedcentral.com/articles/10.1186/s13073-024-01415-3" class="external-link">Mora&amp;Schmidt et al.</a> [@Mora_Schmidt2024].  You can find the detailed code for both manuscripts on our <a href="https://arianemora.github.io/SiRCle_multiomics_integration/#" class="external-link">website</a> or in our <a href="https://github.com/ArianeMora/SiRCle_multiomics_integration" class="external-link">SiRCleManuscript GitHub repository</a>.</p>
<div class="float">
<img src="https://www.biorxiv.org/content/biorxiv/early/2022/07/04/2022.07.02.498058/F6.large.jpg?width=800&amp;height=600&amp;carousel=1" alt="The SiRCle method integrates DNA methylation, RNA-seq and proteomics data at the gene level to deconvolute the association between dysregulation within and across possible regulatory layers (DNA methylation, transcription and/or translation).Based on logical regulatory rules, sircleRCM, genes are grouped in SiRCle clusters based on the layer (DNA methylation, transcription and/or translation) where dys-regulation first occurs. Using the output of sircleRCM, the SiRCle clusters, one can find the primary biological processes altered by applying Over Representation Analysis (ORA) (sircleORA) and the drivers behind it using Transcription Factor (TF) analysis (sircleTF). Lastly, to compare patient’s subsets (e.g. based on stage), we found that integrating across the data layers prior to performing differential analysis and biological enrichment better captures the biological signal. Hence we use a variational autoencoder (VAE) to learn gene-wise relationships across the three data layers to obtain an integrated value for each gene (sircleVAE). Unsing the integarted value we next perform a Mann-Whitney U test to identify genes with a significant integrated difference between the patient’s groups."><div class="figcaption">The SiRCle method integrates DNA methylation, RNA-seq and proteomics data at the gene level to deconvolute the association between dysregulation within and across possible regulatory layers (DNA methylation, transcription and/or translation).Based on logical regulatory rules, <code>sircleRCM</code>, genes are grouped in SiRCle clusters based on the layer (DNA methylation, transcription and/or translation) where dys-regulation first occurs. Using the output of <code>sircleRCM</code>, the SiRCle clusters, one can find the primary biological processes altered by applying Over Representation Analysis (ORA) (<strong><code>sircleORA</code></strong>) and the drivers behind it using Transcription Factor (TF) analysis (<strong><code>sircleTF</code></strong>). Lastly, to compare patient’s subsets (e.g. based on stage), we found that integrating across the data layers prior to performing differential analysis and biological enrichment better captures the biological signal. Hence we use a variational autoencoder (VAE) to learn gene-wise relationships across the three data layers to obtain an integrated value for each gene (<strong><code>sircleVAE</code></strong>). Unsing the integarted value we next perform a Mann-Whitney U test to identify genes with a significant integrated difference between the patient’s groups.</div>
</div>
<p><strong><em>SiRCle</em></strong> uses logical regulatory rules, <code>sircleRCM</code> to group genes in SiRCle clusters based on the data layer (DNA methylation, transcription and/or translation) where dys-regulation first occurs.<br>
Using the output of <code>sircleRCM</code>, the SiRCle clusters, one can find the primary biological processes altered by applying Over Representation Analysis (ORA) and the drivers behind it using Transcription Factor (TF) analysis.<br>
Lastly, to compare patient’s subsets (e.g. based on stage), we found that integrating across the data layers prior to performing differential analysis and biological enrichment better captures the biological signal. Hence we use a variational autoencoder (VAE) to learn gene-wise relationships across the three data layers to obtain an integrated value for each gene (<strong><code>sircleVAE</code></strong>). Using the integrated value we next perform a Mann-Whitney U test to identify genes with a significant integrated difference between the patient’s groups.</p>
</div>
<div class="section level1">
<h1 id="tutorials">Tutorials<a class="anchor" aria-label="anchor" href="#tutorials"></a>
</h1>
<p>We have generated several vignettes showcasing the the usage of SiRCleR using the publicly available datasets as in the manuscript <a href="https://genomemedicine.biomedcentral.com/articles/10.1186/s13073-024-01415-3" class="external-link">Mora&amp;Schmidt et al.</a>, which are included as example data within SiRCleR. You can find those tutorial on the top under the “Tutorials” button, where you can follow a specific user case example and learn more about how to choose the settings including the <code>Background Method</code>, <code>input data thresholds</code> and <code>Regulation Grouping</code>. You can also follow the links below:<br>
* <a href="https://ArianeMora.github.io/SiRCleR/articles/ChooseSettings.html" class="external-link">ChooseSettings</a><br>
* <a href="https://ArianeMora.github.io/SiRCleR/articles/DataAnalysisExample.html" class="external-link">DataAnalysisExample</a></p>
<p>If you are seeking to also apply the variational autoencoder on the SiRCle clustering output, please visit the python package or use reticulate in R. You can find the python package on the <a href="https://github.com/ArianeMora/scircm" class="external-link">Python SiRCle</a>.</p>
</div>
<div class="section level1">
<h1 id="install">Install<a class="anchor" aria-label="anchor" href="#install"></a>
</h1>
<p><strong><code>SiRCleR</code></strong> is an R package. 1. Install Rtools if you haven’t done this yet, using the appropriate version (e.g.<a href="https://cran.r-project.org/bin/windows/Rtools/" class="external-link">windows</a> or <a href="https://cran.r-project.org/bin/macosx/tools/" class="external-link">macOS</a>). 2. Install the latest development version from GitHub with: <strong><code>SiRCleR package</code></strong> direcly in R: <code>devtools::install_github("https://github.com/ArianeMora/SiRCleR")     library(SiRCleR)</code></p>
<div class="section level2">
<h2 id="dependencies">Dependencies<a class="anchor" aria-label="anchor" href="#dependencies"></a>
</h2>
<p>While we have done our best to ensure all the dependencies are documented, if they aren’t please let us know and we will try to resolve them.</p>
</div>
<div class="section level2">
<h2 id="windows-specifications">Windows specifications<a class="anchor" aria-label="anchor" href="#windows-specifications"></a>
</h2>
<p>Note if you are running <strong>Windows</strong> you might have an issue with long paths, which you can resolve in the registry on Windows 10:<br><code>Computer Configuration &gt; Administrative Templates &gt; System &gt; Filesystem &gt; Enable Win32 long paths</code><br>
(If you have a different version of Windows, just google “Long paths fix” and your Windows version)</p>
</div>
<div class="section level2">
<h2 id="questions--issues">Questions &amp; Issues<a class="anchor" aria-label="anchor" href="#questions--issues"></a>
</h2>
<p>Please post questions and issues related to <strong><code>SiRCleR</code></strong> functions on the <code>Issues</code> section of this GitHub repository.</p>
</div>
<div class="section level2">
<h2 id="reproducibility">Reproducibility<a class="anchor" aria-label="anchor" href="#reproducibility"></a>
</h2>
<p>If you want to reproduce the results of of our <a href="https://doi.org/10.1101/2022.07.02.498058" class="external-link">publication</a>, please use the python package version found here: <a href="https://doi.org/10.1101/2022.07.02.498058" class="external-link uri">https://doi.org/10.1101/2022.07.02.498058</a></p>
</div>
</div>
<div class="section level1">
<h1 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h1>
<p>If you use this please cite our manuscript <a href="https://genomemedicine.biomedcentral.com/articles/10.1186/s13073-024-01415-3" class="external-link">Mora&amp;Schmidt et al.</a>. Ariane Mora &amp; Christina Schmidt, Brad Balderson, Christian Frezza &amp; Mikael Bodén. 2024. “SiRCle (Signature Regulatory Clustering) Model Integration Reveals Mechanisms of Phenotype Regulation in Renal Cancer.” Genome Medicine 16 (1): 144. <a href="https://doi.org/10.1186/s13073-024-01415-3" class="external-link uri">https://doi.org/10.1186/s13073-024-01415-3</a>.</p>
</div>

  </main><aside class="col-md-3"><div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Ariane Mora <br><small class="roles"> Author </small>  </li>
<li>Christina Schmidt <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-3867-0881" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Brad Balderson <br><small class="roles"> Author </small>  </li>
<li>Christian Frezza <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-3293-7397" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Mikael Boden <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0003-3548-268X" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

<div class="table-of-contents">
<h2 data-toc-skip>Table of contents</h2>
<ul class="list-unstyled">
<li><nav id="toc"></nav></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Ariane Mora, Christina Schmidt, Brad Balderson, Christian Frezza, Mikael Boden.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
