<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>sircleRCM_MRP — sircleRCM_MRP • SiRCleR</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="sircleRCM_MRP — sircleRCM_MRP"><meta name="description" content="Computes the regulatory clustering model (RCM) using the SiRCle regulatory rules for DNA-methylation, RNAseq and Proteomics data layers (MRP)."><meta property="og:description" content="Computes the regulatory clustering model (RCM) using the SiRCle regulatory rules for DNA-methylation, RNAseq and Proteomics data layers (MRP)."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">SiRCleR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-tutorials" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true"><span class="fa fa-book-open"></span> Tutorials</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-tutorials"><li><a class="dropdown-item" href="../articles/ChooseSettings.html">ChooseSettings</a></li>
    <li><a class="dropdown-item" href="../articles/DataAnalysisExample.html">TutorialR</a></li>
  </ul></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html"><span class="fa fa-solid fa-rocket"></span> Reference</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><a class="external-link nav-link" href="https://github.com/ArianeMora/SiRCleR"><span class="fa fa-brands fa-github"></span></a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/ArianeMora/scircm"><span class="fa fa-brands fa-python"></span></a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://genomemedicine.biomedcentral.com/articles/10.1186/s13073-024-01415-3"><span class="fa fa-solid fa-dna"></span> Manuscript</a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>sircleRCM_MRP</h1>

      <div class="d-none name"><code>sircleRCM_MRP.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Computes the regulatory clustering model (RCM) using the SiRCle regulatory rules for DNA-methylation, RNAseq and Proteomics data layers (MRP).</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">sircleRCM_MRP</span><span class="op">(</span></span>
<span>  <span class="va">methFile</span>,</span>
<span>  <span class="va">rnaFile</span>,</span>
<span>  <span class="va">protFile</span>,</span>
<span>  <span class="va">geneID</span>,</span>
<span>  rnaValueCol <span class="op">=</span> <span class="st">"Log2FC"</span>,</span>
<span>  rnaPadjCol <span class="op">=</span> <span class="st">"padj"</span>,</span>
<span>  methValueCol <span class="op">=</span> <span class="st">"Diff"</span>,</span>
<span>  methPadjCol <span class="op">=</span> <span class="st">"padj"</span>,</span>
<span>  proteinValueCol <span class="op">=</span> <span class="st">"Log2FC"</span>,</span>
<span>  proteinPadjCol <span class="op">=</span> <span class="st">"padj"</span>,</span>
<span>  rna_padj_cutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  prot_padj_cutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  meth_padj_cutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  rna_FC_cutoff <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  prot_FC_cutoff <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  meth_Diff_cutoff <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  <span class="va">backgroundMethod</span>,</span>
<span>  OutputFileName <span class="op">=</span> <span class="st">"Sircle_RCM"</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-methfile">methFile<a class="anchor" aria-label="anchor" href="#arg-methfile"></a></dt>
<dd><p>DF for your DNA methylation data (results from differential methylation analysis)</p></dd>


<dt id="arg-rnafile">rnaFile<a class="anchor" aria-label="anchor" href="#arg-rnafile"></a></dt>
<dd><p>DF for your RNAseq data (results from DeSeq2 and also your normalised expression counts)</p></dd>


<dt id="arg-geneid">geneID<a class="anchor" aria-label="anchor" href="#arg-geneid"></a></dt>
<dd><p>Column name of geneId this MUST BE THE SAME in each of your protein, RNAseq and DNAmethylation files (we join on this)</p></dd>


<dt id="arg-rnavaluecol">rnaValueCol<a class="anchor" aria-label="anchor" href="#arg-rnavaluecol"></a></dt>
<dd><p>Column name of RNA value in rnaFile</p></dd>


<dt id="arg-rnapadjcol">rnaPadjCol<a class="anchor" aria-label="anchor" href="#arg-rnapadjcol"></a></dt>
<dd><p>Column name of RNA p adjusted value in rnaFile</p></dd>


<dt id="arg-methvaluecol">methValueCol<a class="anchor" aria-label="anchor" href="#arg-methvaluecol"></a></dt>
<dd><p>Column name of Methylation difference value in methFile</p></dd>


<dt id="arg-methpadjcol">methPadjCol<a class="anchor" aria-label="anchor" href="#arg-methpadjcol"></a></dt>
<dd><p>Column name of Methylation p adjusted value in methFile</p></dd>


<dt id="arg-proteinvaluecol">proteinValueCol<a class="anchor" aria-label="anchor" href="#arg-proteinvaluecol"></a></dt>
<dd><p>Column name of protein log fold change in proteinFile</p></dd>


<dt id="arg-proteinpadjcol">proteinPadjCol<a class="anchor" aria-label="anchor" href="#arg-proteinpadjcol"></a></dt>
<dd><p>Column name of protein p adjusted value in proteinFile</p></dd>


<dt id="arg-backgroundmethod">backgroundMethod<a class="anchor" aria-label="anchor" href="#arg-backgroundmethod"></a></dt>
<dd><p><em>Optional: </em> Background methods: "P|(M&amp;R)", "P|M|R", "P|R", "P&amp;R", "P&amp;M&amp;R", "(P&amp;M)|(P&amp;R)|(M&amp;R)", "(P&amp;M)|(P&amp;R)" or "*". For details on the methods, please check https://github.com/ArianeMora/SiRCleR/blob/main/vignettes/SiRCle_RCM_Notebook.Rmd <strong>Default="P|(M&amp;R)"</strong></p></dd>


<dt id="arg-proteinfile">proteinFile<a class="anchor" aria-label="anchor" href="#arg-proteinfile"></a></dt>
<dd><p>DF of you Protein data (results from DeSeq2 and also your normalised expression counts)</p></dd>


<dt id="arg-rnapadjcutoff">rnaPadjCutoff<a class="anchor" aria-label="anchor" href="#arg-rnapadjcutoff"></a></dt>
<dd><p><em>Optional: </em>Padjusted cutoff for RNAseq data <strong>Default=0.05</strong></p></dd>


<dt id="arg-rnalogfccutoff">rnaLogFCCutoff<a class="anchor" aria-label="anchor" href="#arg-rnalogfccutoff"></a></dt>
<dd><p><em>Optional: </em> LogFoldchange cutoff for RNAseq data <strong>Default=0.5</strong></p></dd>


<dt id="arg-proteinpadjcutoff">proteinPadjCutoff<a class="anchor" aria-label="anchor" href="#arg-proteinpadjcutoff"></a></dt>
<dd><p><em>Optional: </em> Padjusted cutoff for Protein data <strong>Default=0.05</strong></p></dd>


<dt id="arg-proteinvaluecutoff">proteinValueCutoff<a class="anchor" aria-label="anchor" href="#arg-proteinvaluecutoff"></a></dt>
<dd><p><em>Optional: </em> LogFoldchange cutoff for Protein data <strong>Default=0.3</strong></p></dd>


<dt id="arg-methpadjcutoff">methPadjCutoff<a class="anchor" aria-label="anchor" href="#arg-methpadjcutoff"></a></dt>
<dd><p><em>Optional: </em> Padjusted cutoff for DNA methylation data <strong>Default=0.05</strong></p></dd>


<dt id="arg-methdiffcutoff">methDiffCutoff<a class="anchor" aria-label="anchor" href="#arg-methdiffcutoff"></a></dt>
<dd><p><em>Optional: </em> DNA Methylation difference cutoff for DNA methylation <strong>Default=10</strong></p></dd>


<dt id="arg-outputfilename">outputFileName<a class="anchor" aria-label="anchor" href="#arg-outputfilename"></a></dt>
<dd><p><em>Optional: </em> Output filename <strong>Default="SiRCle_RCM"</strong></p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>DF with input data combined by geneID and SiRCle Clusters</p>
    </div>

  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Ariane Mora, Christina Schmidt, Brad Balderson, Christian Frezza, Mikael Boden.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

