<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>SiRCle RCM Rules • SiRCleR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="SiRCle RCM Rules">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">SiRCleR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-tutorials" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true"><span class="fa fa-book-open"></span> Tutorials</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-tutorials">
<li><a class="dropdown-item" href="../articles/ChooseSettings.html">ChooseSettings</a></li>
    <li><a class="dropdown-item" href="../articles/DataAnalysisExample.html">TutorialR</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html"><span class="fa fa-solid fa-rocket"></span> Reference</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/ArianeMora/SiRCleR"><span class="fa fa-brands fa-github"></span></a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/ArianeMora/scircm"><span class="fa fa-brands fa-python"></span></a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://genomemedicine.biomedcentral.com/articles/10.1186/s13073-024-01415-3"><span class="fa fa-solid fa-dna"></span> Manuscript</a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">



<script src="ChooseSettings_files/kePrint-0.0.1/kePrint.js"></script><link href="ChooseSettings_files/lightable-0.0.1/lightable.css" rel="stylesheet">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>SiRCle RCM Rules</h1>
                        <h4 data-toc-skip class="author">Ariane
Mora</h4>
                                    <h4 data-toc-skip class="author">Christina
Schmidt</h4>
                        
      

      <div class="d-none name"><code>ChooseSettings.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="sircle-regulatory-clustering-model">
<strong><em>SiRCle Regulatory Clustering Model</em></strong><a class="anchor" aria-label="anchor" href="#sircle-regulatory-clustering-model"></a>
</h2>
<div class="progress progress-striped active">
<div class="progress-bar" style="width: 100%">

</div>
</div>
<br><br>
This notebook is meant as a guide for choosing the settings for the
Signature Regulatory Clustering (SiRCle) dependent on the biological
question of interest. This guide is divided into three main chapters
that give an overview on the SiRcle method, each covering one of the
main settings in the <code>sircleRCM() functions</code> (Regulatory
Clustering Method) namely the “Background method (BG)”, the “Regulation
Grouping (RG)” and the thresholds/cutoffs of the input data.<br><br><strong><em>sircleRCM_MRP()</em></strong><br>
The original Regulatory Clustering Method (RCM)
<code><a href="../reference/sircleRCM_MRP.html">sircleRCM_MRP()</a></code> was developed to perform biologically
meaningful clustering integrating DNA methylation, mRNA, and protein
data at the gene level. Based on the differential analysis (e.g. Tumour
versus Normal), we defined three states, namely positive, negative or
unchanged, for each gene and data layer. Given that the protein
expression is the data layer closest to the phenotype, yet at the same
time has the lowest coverage, “No Change” is subdivided into four states
for the protein data layer, “Not Detected”, “Not Significant”,
“significant negative” and “significant positive”. For more details and
explanation check out the publication from <a href="https://genomemedicine.biomedcentral.com/articles/10.1186/s13073-024-01415-3" class="external-link">Mora&amp;Schmidt
et al.</a> <span class="citation">(Mora et al. 2024)</span>.<br><br><center>
<div class="float">
<img src="sircleRCM_MRP.png" style="width:90.0%" alt="Fig. 1: Overview of SiRCle Regulatory Clustering. a) Alluvial plot showing the ordered series of three 3-state-3-state-6-state transitions between the data layers, a total of 54 possible “flows”. b) Biological meaning the flows between the data layers will conveye."><div class="figcaption">
<strong>Fig. 1:</strong> Overview of SiRCle
Regulatory Clustering. a) Alluvial plot showing the ordered series of
three 3-state-3-state-6-state transitions between the data layers, a
total of 54 possible “flows”. b) Biological meaning the flows between
the data layers will conveye.</div>
</div>
</center>
<br><br><strong><em>sircleRCM_RP()</em></strong><br>
It is likely that one only obtained data for the RNA and protein data
layer. This is the reason why we also offer the option of the
<code><a href="../reference/sircleRCM_RP.html">sircleRCM_RP()</a></code> function, which only uses RNAseq and
proteomics as input. The above principle stays the same, yet here we one
have two 3-state-6-state transitions between the data layers, a total of
18 possible “flows”.<br><br><center>
<div class="float">
<img src="sircleRCM_RP.png" style="width:40.0%" alt="Fig. 2: Overview of SiRCle Regulatory Clustering without the DNA-Methylation data layer."><div class="figcaption">
<strong>Fig. 2:</strong> Overview of SiRCle
Regulatory Clustering without the DNA-Methylation data layer.</div>
</div>
</center>
<p><br><br><strong><em>sircleRCM_2Cond()</em></strong><br>
Lastly, to enable the application of SiRCle RCM to any two data layers
of choice, we provide the <code><a href="../reference/sircleRCM_2Cond.html">sircleRCM_2Cond()</a></code> function. Here
one can either input two different data layers, but also the same data
layers (e.g. RNAseq comparing KO versus WT in hypoxia and in normoxia).
In the above options we take into account that the protein expression is
the data layer closest to the phenotype, yet at the same time has the
lowest coverage. Here we want to give the full flexibility to the user
and hence we have a two 6-state-6-state transitions between the data
layers, a total of 36 possible “flows”.<br><br>
In the next chapters we will discuss how to choose the “Background
method (BG)”, the “Regulation Grouping (RG)” and the thresholds/cutoffs
of the input data.<br><br></p>
<div class="progress progress-striped active">
<div class="progress-bar" style="width: 100%">

</div>
</div>
</div>
<div class="section level2">
<h2 id="background-method-bg">
<strong><em>1. Background method (BG)</em></strong><a class="anchor" aria-label="anchor" href="#background-method-bg"></a>
</h2>
<div class="progress progress-striped active">
<div class="progress-bar" style="width: 100%">

</div>
</div>
<p>The Background setting defines, which features will be considered for
the clusters. For example if you have two data layers, RNAseq and
proteomics, you could include only features (= genes) that are detected
in both data layers, removing the rest of the features.</p>
<div class="section level3">
<h3 id="three-data-layers-methylation-rnaseq-and-proteomics">
<strong><em>Three data layers: Methylation, RNAseq and
Proteomics</em></strong><a class="anchor" aria-label="anchor" href="#three-data-layers-methylation-rnaseq-and-proteomics"></a>
</h3>
<p><strong><em>1.1. - 1.7.</em></strong> goes from most restrictive to
least restrictive. Hence the chosen background method will define the
number of genes included in the <code>sircleRCM</code> input. Given that
the proteomics data has the lowest coverage of the three input data
layers (Proteomics, RNAseq and DNA-methylation), the proteomics data
layer will have the lowest number of detected genes (=features) and
hence has the biggest impact on the number of genes included in the
<code>sircleRCM</code> input.<br><br><strong><em>1.1. P&amp;M&amp;R</em></strong><br>
Most stringend background setting and will lead to a small number of
genes that is mainly dependent on the proteins detected (proteomics data
layer), since proteomics has the lowest coverage of the three
omics.<br><br><strong><em>1.2. P&amp;R</em></strong><br>
This background ensures to detect a gene in Proteomics and RNAseq.
Similar to 1.1, yet less stringent as the gene does not need to be
detected in DNA-Methylation.<br><br><strong><em>1.3. P|(M&amp;R)</em></strong><br>
Focus is on protein expression. Even though it is unlikely that
P&amp;M&amp;R (1.1.) or P&amp;R (1.2.) excludes many proteins, a protein
could be excluded if it is not detected on the RNAseq and/or
DNA-methylation data layer. Hence, this background P | (M&amp;R) (1.2.)
ensures to include all of the detected proteins (= all features on the
proteomics data layer). On the other hand it will also include genes
that are not detected on the protein layer (proteomics data), but on the
mRNA (RNAseq data) and on for DNA-methylation (DNA-methylation
data).<br><br><strong><em>1.4. (P&amp;M)|(P&amp;R)</em></strong><br>
This background ensures to detect a gene in Proteomics and one of the
other two data layers. Similar to 1.1, yet less stringent as the gene
does not need to be detected in RNAseq and DNA-Methylation.<br><br><strong><em>1.5. (P&amp;M)|(P&amp;R)|(M&amp;R)</em></strong><br>
This background ensures to detect a gene in at least two of the three
data layers. Hence, the focus moves away from the protein layer
(proteomics data layer) and the number of genes that are not detected on
the proteomics data layer will increase in comparison to P &amp; M &amp;
R (1.1.) and P | (M&amp;R) (1.2.).<br><br><strong><em>1.6. P|R</em></strong><br>
A gene will be included in the SiRCle input if it has been detected on
one of the two data layers (proteomics or RNAseq).<br><br><strong><em>1.7. P|M|R</em></strong><br>
Least stringent background method, since a gene will be included in the
SiRCle input if it has been detected on one of the three data layers
(proteomics, RNAseq or DNA-methylation).<br></p>
</div>
<div class="section level3">
<h3 id="two-data-layers-rnaseq-and-proteomics">
<strong><em>Two data layers: RNAseq and
Proteomics</em></strong><a class="anchor" aria-label="anchor" href="#two-data-layers-rnaseq-and-proteomics"></a>
</h3>
<p>As discussed above, often only RNAseq and Proteomics is conducted,
and hence there is the <code><a href="../reference/sircleRCM_RP.html">sircleRCM_RP()</a></code> function focusing on
only those two data layers not requiring the DNA-methylation data. In
turn the background methods are the following from <strong><em>1.1. -
1.4.</em></strong> from most restrictive to least restrictive:</p>
<p><strong><em>1.1. P&amp;R</em></strong><br>
Most stringend background setting and will lead to a small number of
genes that is mainly dependent on the proteins detected (proteomics data
layer), since proteomics has the lower coverage compared to
RNAseq.<br><br><strong><em>1.2. P</em></strong><br>
Focus is on protein expression. Even though it is unlikely that P&amp;R
(1.1.) excludes many proteins, a protein could be excluded if it is not
detected on the RNAseq data layer. Hence, this background ensures to
include all of the detected proteins (= all features on the proteomics
data layer).<br><br><strong><em>1.3. R</em></strong><br>
Focus is on mRNA expression. P&amp;R (1.1.) will exclude many genes from
the mRNA data layer, hence this background ensures to include all of the
detected mRNAs (= all features on the RNAseq data layer).<br><br><strong><em>1.4. P|R</em></strong><br>
Least stringent background method, since a gene will be included in the
SiRCle input if it has been detected on one of the two data layers
(proteomics or RNAseq).<br></p>
</div>
<div class="section level3">
<h3 id="two-data-sets">
<strong><em>Two data sets</em></strong><a class="anchor" aria-label="anchor" href="#two-data-sets"></a>
</h3>
<p>Here one can either input two different data layers or two different
conditions of the same data layer. In turn the background methods are
the following from <strong><em>1.1. - 1.4.</em></strong> from most
restrictive to least restrictive:<br><br><strong><em>1.1. C1&amp;C2</em></strong><br>
Most stringend background setting and will lead to a small number of
genes.<br><br><strong><em>1.2. C1</em></strong><br>
Focus is on The gene expression of Condition 1 (C1).<br><br><strong><em>1.3. C2</em></strong><br>
Focus is on The gene expression of Condition 2 (C2).<br><br><strong><em>1.4. C1|C2</em></strong><br>
Least stringent background method, since a gene will be included in the
SiRCle input if it has been detected on one of the two conditions.<br><br></p>
<div class="progress progress-striped active">
<div class="progress-bar" style="width: 100%">

</div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="input-data-threshold">
<strong><em>2. Input data threshold</em></strong><a class="anchor" aria-label="anchor" href="#input-data-threshold"></a>
</h2>
<div class="progress progress-striped active">
<div class="progress-bar" style="width: 100%">

</div>
</div>
<p>Here we can set two different thresholds, one for the differential
expression, which would mostly be <code>Log2FC</code> for the RNAseq and
proteomics data layer and
<code>Differential Methylated Regions (DMR)</code> for DNA-methylation,
and one for the <code>significance</code> (e.g. p.adj). This will define
if a feature (= e.g. gene) is assigned into:<br><br>
1. <strong><em>“UP”</em></strong>, which means a feature is
significantly up-regulated in the underlying comparison.<br><br>
2. <strong><em>“DOWN”</em></strong>, which means a feature is
significantly down-regulated in the underlying comparison.<br><br>
3. <strong><em>“No Change”</em></strong>, which means a feature does not
change significantly in the underlying comparison and/or is not defined
as up-regulated/down-regulated based on the Log2FC threshold
chosen.<br><br><br>
Given that the protein expression is the data layer closest to the
phenotype, yet at the same time has the lowest coverage, “No Change” is
subdivided into four states for the protein data layer:<br><br>
1. <strong><em>“Not Detected”</em></strong>, which means a feature is
not detected in the underlying data layer.<br><br>
2. <strong><em>“Not Significant”</em></strong>, which means a feature is
not significant in the underlying comparison.<br><br>
3. <strong><em>“Significant negative”</em></strong>, which means a
feature is significant in the underlying comparison and the differential
expression is positive, yet does not meet the threshold set for “UP”
(e.g. Log2FC &gt;1 = “UP” and we have a significant Log2FC=0.8).<br><br>
4. <strong><em>“Significant positive”</em></strong>, which means a
feature is significant in the underlying comparison and the differential
expression is positive, yet does not meet the threshold set for
“DOWN”.<br><br><br>
This definition is done individually for each data layer,(proteomics,
RNAseq, DNA-methylation) and will impact in which SiRCle cluster a gene
is sorted into. The latter will be discussed in detail below (see “3.
Regulation Grouping (RG)”).</p>
<div class="progress progress-striped active">
<div class="progress-bar" style="width: 100%">

</div>
</div>
</div>
<div class="section level2">
<h2 id="regulation-grouping-rg">
<strong><em>3. Regulation Grouping (RG)</em></strong><a class="anchor" aria-label="anchor" href="#regulation-grouping-rg"></a>
</h2>
<div class="progress progress-striped active">
<div class="progress-bar" style="width: 100%">

</div>
</div>
Based on the background method (BG) we have defined the genes that are
included in the SiRCle input and based on our cut-offs (Log2FC and
p-adjusted value) we defined the change (“UP”, “DOWN”, “No Change”) a
gene is assigned to.<br>
The results of the SirCle RCM includes different Regulation Groupings
(RG), which define the SiRCle cluster a gene will be assigned to,
therebye summarizing the flows into a smaller amount of SiRCle clusters.
In brief, each SiRCle cluster reflects the regulation(s) that ultimately
result in the protein expression and the names assigned to the SiRCle
clusters reflect the biological meaning, namely “Enhancing (+1)” or
“Suppressing (-1)”, of the regulation (see Fig. 1b). Hence, a regulation
can happen between DNA methylation and mRNA expression (MR) and/or
between mRNA expression and protein expression (RP). Each change of
regulation that happens between the data layers will define the SiRCle
cluster’s name. For example, if a gene is hypermethylated, has a
decrease in mRNA expression, and displays a decrease in protein
expression, we can likely conclude dysregulation first occurred on the
DNA methylation layer, meaning this gene is suppressed via
Methylation-Driven Suppression (MDS).<br><br><center>
<div class="float">
<img src="sircleRCM_MRP_b.png" style="width:40.0%" alt="Fig. 1b: Biological meaning the flows between the data layers will conveye."><div class="figcaption">
<strong>Fig. 1b:</strong> Biological meaning the
flows between the data layers will conveye.</div>
</div>
</center>
<br><br>
In Figure 1b, we see three different plots and in each plot we have 9
flows connecting the three data layers (DNA methylation, mRNA expression
and Protein expression). This gives a total of 27 possible flows.<br>
Given that our regulatory rules have the aim to reflects the
regulation(s) that ultimately result in the protein expression, it
becomes essential to deal with proteins assigned to “No Change”, since
“No Change” on the protein level can mean:<br>
1. <strong><em>“Not Detected”</em></strong><br>
2. <strong><em>“Not Significant”</em></strong><br>
3. <strong><em>“Significant negative”</em></strong><br>
4. <strong><em>“Significant positive”</em></strong><br><br>
Depending on the “RG” chosen by the user, different granularities of the
regulations are considered, which can lead to SiRCle cluster names
reflecting one (MR or RP) or two regulations (MR+RP) (see Table
1).<br><strong><em>3.1. RG1: All</em></strong><br>
Column <code>RG1_All</code>, which includes all 54 possible flows (=
ordered series of three 3-state-3-state-6-state transitions between the
data layers) as the output and does not summarizes multiple flows into a
SiRCle cluster.<br><br><strong><em>3.2. RG2: Focus Changes</em></strong><br>
Column <code>RG2_Changes</code>, which summarizes the 54 possible flows
into 10 SiRCle clusters taking into account any changes between the data
layers (Table 1). Here we focus on changes and hence reflect any small
change between the data layers by including in in the SiRCle
cluster.<br><br><strong><em>3.3. RG3: Focus Translation</em></strong><br>
Column <code>RG3_Protein</code>, which only take into account regulation
between mRNA expression and protein expression (RP). Meaning even if
tehre has been also a regulation between the DNA-methylation layer and
the RNA layer, this is ignored and the genes are just sorted based on
the secondary regulation on the translational level.<br><br><strong><em>3.4 RG4: Focus Detection</em></strong><br>
Column <code>RG4_Detection</code>, will not take into account changes
between mRNA and protein layer if a protein has not been detected. In
this case only the regulation between DNA-methylation and RNA is taken
into account to define the SiRCle cluster. Of course, if one choses any
Background method that enforces the protein to be detected
(e.g. M&amp;R&amp;P), Column <code>RG4_Detection</code> will include the
same as RG3_Protein as the flow trough “Not Detected” can not
occur.<br><br><div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:600px; overflow-x: scroll; width:100%; ">
<table class="table table table-striped" style="width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
Table 1: Regulatory Grouping (RG) 1-4 for <code><a href="../reference/sircleRCM_MRP.html">sircleRCM_MRP()</a></code>
function
</caption>
<thead><tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Methylation
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
RNA-seq
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Proteomics
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Proteomics_Detection
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
RG1_All
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
RG2_Changes
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
RG3_Protein
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
RG4_Detection
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
Hypermethylation
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Hypermethylation + DOWN + DOWN
</td>
<td style="text-align:left;">
MDS
</td>
<td style="text-align:left;">
MDS
</td>
<td style="text-align:left;">
MDS
</td>
</tr>
<tr>
<td style="text-align:left;">
Hypomethylation
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Hypomethylation + DOWN + DOWN
</td>
<td style="text-align:left;">
TPDS
</td>
<td style="text-align:left;">
TPDS
</td>
<td style="text-align:left;">
TPDS
</td>
</tr>
<tr>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
No Change + DOWN + DOWN
</td>
<td style="text-align:left;">
TPDS
</td>
<td style="text-align:left;">
TPDS
</td>
<td style="text-align:left;">
TPDS
</td>
</tr>
<tr>
<td style="text-align:left;">
Hypermethylation
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Hypermethylation + No Change + DOWN
</td>
<td style="text-align:left;">
TMDS
</td>
<td style="text-align:left;">
TMDS
</td>
<td style="text-align:left;">
TMDS
</td>
</tr>
<tr>
<td style="text-align:left;">
Hypomethylation
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Hypomethylation + No Change + DOWN
</td>
<td style="text-align:left;">
TMDS
</td>
<td style="text-align:left;">
TMDS
</td>
<td style="text-align:left;">
TMDS
</td>
</tr>
<tr>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
No Change + No Change + DOWN
</td>
<td style="text-align:left;">
TMDS
</td>
<td style="text-align:left;">
TMDS
</td>
<td style="text-align:left;">
TMDS
</td>
</tr>
<tr>
<td style="text-align:left;">
Hypermethylation
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Hypermethylation + UP + DOWN
</td>
<td style="text-align:left;">
TPDE+TMDS
</td>
<td style="text-align:left;">
TMDS
</td>
<td style="text-align:left;">
TPDE+TMDS
</td>
</tr>
<tr>
<td style="text-align:left;">
Hypomethylation
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Hypomethylation + UP + DOWN
</td>
<td style="text-align:left;">
MDE+TMDS
</td>
<td style="text-align:left;">
TMDS
</td>
<td style="text-align:left;">
MDE+TMDS
</td>
</tr>
<tr>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
No Change + UP + DOWN
</td>
<td style="text-align:left;">
TPDE+TMDS
</td>
<td style="text-align:left;">
TMDS
</td>
<td style="text-align:left;">
TPDE+TMDS
</td>
</tr>
<tr>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
No Change + No Change + No Change
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Hypermethylation
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
Not detected
</td>
<td style="text-align:left;">
Hypermethylation + DOWN + Not detected
</td>
<td style="text-align:left;">
MDS+TMDE
</td>
<td style="text-align:left;">
MDS
</td>
<td style="text-align:left;">
MDS
</td>
</tr>
<tr>
<td style="text-align:left;">
Hypomethylation
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
Not detected
</td>
<td style="text-align:left;">
Hypomethylation + DOWN + Not detected
</td>
<td style="text-align:left;">
TPDS+TMDE
</td>
<td style="text-align:left;">
TPDS
</td>
<td style="text-align:left;">
TPDS
</td>
</tr>
<tr>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
Not detected
</td>
<td style="text-align:left;">
No Change + DOWN + Not detected
</td>
<td style="text-align:left;">
TPDS+TMDE
</td>
<td style="text-align:left;">
TPDS
</td>
<td style="text-align:left;">
TPDS
</td>
</tr>
<tr>
<td style="text-align:left;">
Hypermethylation
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
Not detected
</td>
<td style="text-align:left;">
Hypermethylation + No Change + Not detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Hypomethylation
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
Not detected
</td>
<td style="text-align:left;">
Hypomethylation + No Change + Not detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
Not detected
</td>
<td style="text-align:left;">
No Change + No Change + Not detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Hypermethylation
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
Not detected
</td>
<td style="text-align:left;">
Hypermethylation + UP + Not detected
</td>
<td style="text-align:left;">
TPDE+TMDS
</td>
<td style="text-align:left;">
TPDE
</td>
<td style="text-align:left;">
TPDE
</td>
</tr>
<tr>
<td style="text-align:left;">
Hypomethylation
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
Not detected
</td>
<td style="text-align:left;">
Hypomethylation + UP + Not detected
</td>
<td style="text-align:left;">
MDE+TMDS
</td>
<td style="text-align:left;">
MDE
</td>
<td style="text-align:left;">
MDE
</td>
</tr>
<tr>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
Not detected
</td>
<td style="text-align:left;">
No Change + UP + Not detected
</td>
<td style="text-align:left;">
TPDE+TMDS
</td>
<td style="text-align:left;">
TPDE
</td>
<td style="text-align:left;">
TPDE
</td>
</tr>
<tr>
<td style="text-align:left;">
Hypermethylation
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
Not significant
</td>
<td style="text-align:left;">
Hypermethylation + DOWN + Not significant
</td>
<td style="text-align:left;">
MDS+TMDE
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
MDS+TMDE
</td>
</tr>
<tr>
<td style="text-align:left;">
Hypomethylation
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
Not significant
</td>
<td style="text-align:left;">
Hypomethylation + DOWN + Not significant
</td>
<td style="text-align:left;">
TPDS+TMDE
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
TPDS+TMDE
</td>
</tr>
<tr>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
Not significant
</td>
<td style="text-align:left;">
No Change + DOWN + Not significant
</td>
<td style="text-align:left;">
TPDS+TMDE
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
TPDS+TMDE
</td>
</tr>
<tr>
<td style="text-align:left;">
Hypermethylation
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
Not significant
</td>
<td style="text-align:left;">
Hypermethylation + No Change + Not significant
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Hypomethylation
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
Not significant
</td>
<td style="text-align:left;">
Hypomethylation + No Change + Not significant
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
Not significant
</td>
<td style="text-align:left;">
No Change + No Change + Not significant
</td>
<td style="text-align:left;">
TPDS+TMDE
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Hypermethylation
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
Not significant
</td>
<td style="text-align:left;">
Hypermethylation + UP + Not significant
</td>
<td style="text-align:left;">
TPDE+TMDS
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
TPDE+TMDS
</td>
</tr>
<tr>
<td style="text-align:left;">
Hypomethylation
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
Not significant
</td>
<td style="text-align:left;">
Hypomethylation + UP + Not significant
</td>
<td style="text-align:left;">
MDE+TMDS
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
MDE+TMDS
</td>
</tr>
<tr>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
Not significant
</td>
<td style="text-align:left;">
No Change + UP + Not significant
</td>
<td style="text-align:left;">
TPDE+TMDS
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
TPDE+TMDS
</td>
</tr>
<tr>
<td style="text-align:left;">
Hypermethylation
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
Significant negative
</td>
<td style="text-align:left;">
Hypermethylation + DOWN + Significant negative
</td>
<td style="text-align:left;">
MDS+TMDE
</td>
<td style="text-align:left;">
MDS
</td>
<td style="text-align:left;">
MDS
</td>
</tr>
<tr>
<td style="text-align:left;">
Hypomethylation
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
Significant negative
</td>
<td style="text-align:left;">
Hypomethylation + DOWN + Significant negative
</td>
<td style="text-align:left;">
TPDS+TMDE
</td>
<td style="text-align:left;">
TPDS
</td>
<td style="text-align:left;">
TPDS
</td>
</tr>
<tr>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
Significant negative
</td>
<td style="text-align:left;">
No Change + DOWN + Significant negative
</td>
<td style="text-align:left;">
TPDS+TMDE
</td>
<td style="text-align:left;">
TPDS
</td>
<td style="text-align:left;">
TPDS
</td>
</tr>
<tr>
<td style="text-align:left;">
Hypermethylation
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
Significant negative
</td>
<td style="text-align:left;">
Hypermethylation + No Change + Significant negative
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Hypomethylation
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
Significant negative
</td>
<td style="text-align:left;">
Hypomethylation + No Change + Significant negative
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Hypermethylation
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
Significant negative
</td>
<td style="text-align:left;">
Hypermethylation + UP + Significant negative
</td>
<td style="text-align:left;">
TPDE+TMDS
</td>
<td style="text-align:left;">
TMDS
</td>
<td style="text-align:left;">
TPDE+TMDS
</td>
</tr>
<tr>
<td style="text-align:left;">
Hypomethylation
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
Significant negative
</td>
<td style="text-align:left;">
Hypomethylation + UP + Significant negative
</td>
<td style="text-align:left;">
MDE+TMDS
</td>
<td style="text-align:left;">
TMDS
</td>
<td style="text-align:left;">
MDE+TMDS
</td>
</tr>
<tr>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
Significant negative
</td>
<td style="text-align:left;">
No Change + UP + Significant negative
</td>
<td style="text-align:left;">
TPDE+TMDS
</td>
<td style="text-align:left;">
TMDS
</td>
<td style="text-align:left;">
TPDE+TMDS
</td>
</tr>
<tr>
<td style="text-align:left;">
Hypermethylation
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
Significant positive
</td>
<td style="text-align:left;">
Hypermethylation + DOWN + Significant positive
</td>
<td style="text-align:left;">
MDS+TMDE
</td>
<td style="text-align:left;">
TMDE
</td>
<td style="text-align:left;">
MDS+TMDE
</td>
</tr>
<tr>
<td style="text-align:left;">
Hypomethylation
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
Significant positive
</td>
<td style="text-align:left;">
Hypomethylation + DOWN + Significant positive
</td>
<td style="text-align:left;">
TPDS+TMDE
</td>
<td style="text-align:left;">
TMDE
</td>
<td style="text-align:left;">
TPDS+TMDE
</td>
</tr>
<tr>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
Significant positive
</td>
<td style="text-align:left;">
No Change + DOWN + Significant positive
</td>
<td style="text-align:left;">
TPDS+TMDE
</td>
<td style="text-align:left;">
TMDE
</td>
<td style="text-align:left;">
TPDS+TMDE
</td>
</tr>
<tr>
<td style="text-align:left;">
Hypermethylation
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
Significant positive
</td>
<td style="text-align:left;">
Hypermethylation + No Change + Significant positive
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Hypomethylation
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
Significant positive
</td>
<td style="text-align:left;">
Hypomethylation + No Change + Significant positive
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Hypermethylation
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
Significant positive
</td>
<td style="text-align:left;">
Hypermethylation + UP + Significant positive
</td>
<td style="text-align:left;">
TPDE+TMDS
</td>
<td style="text-align:left;">
TPDE
</td>
<td style="text-align:left;">
TPDE
</td>
</tr>
<tr>
<td style="text-align:left;">
Hypomethylation
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
Significant positive
</td>
<td style="text-align:left;">
Hypomethylation + UP + Significant positive
</td>
<td style="text-align:left;">
MDE+TMDS
</td>
<td style="text-align:left;">
MDE
</td>
<td style="text-align:left;">
MDE
</td>
</tr>
<tr>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
Significant positive
</td>
<td style="text-align:left;">
No Change + UP + Significant positive
</td>
<td style="text-align:left;">
TPDE+TMDS
</td>
<td style="text-align:left;">
TPDE
</td>
<td style="text-align:left;">
TPDE
</td>
</tr>
<tr>
<td style="text-align:left;">
Hypermethylation
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Hypermethylation + DOWN + UP
</td>
<td style="text-align:left;">
MDS+TMDE
</td>
<td style="text-align:left;">
TMDE
</td>
<td style="text-align:left;">
MDS+TMDE
</td>
</tr>
<tr>
<td style="text-align:left;">
Hypomethylation
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Hypomethylation + DOWN + UP
</td>
<td style="text-align:left;">
TPDS+TMDE
</td>
<td style="text-align:left;">
TMDE
</td>
<td style="text-align:left;">
TPDS+TMDE
</td>
</tr>
<tr>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
No Change + DOWN + UP
</td>
<td style="text-align:left;">
TPDS+TMDE
</td>
<td style="text-align:left;">
TMDE
</td>
<td style="text-align:left;">
TPDS+TMDE
</td>
</tr>
<tr>
<td style="text-align:left;">
Hypermethylation
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Hypermethylation + No Change + UP
</td>
<td style="text-align:left;">
TMDE
</td>
<td style="text-align:left;">
TMDE
</td>
<td style="text-align:left;">
TMDE
</td>
</tr>
<tr>
<td style="text-align:left;">
Hypomethylation
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Hypomethylation + No Change + UP
</td>
<td style="text-align:left;">
TMDE
</td>
<td style="text-align:left;">
TMDE
</td>
<td style="text-align:left;">
TMDE
</td>
</tr>
<tr>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
No Change + No Change + UP
</td>
<td style="text-align:left;">
TMDE
</td>
<td style="text-align:left;">
TMDE
</td>
<td style="text-align:left;">
TMDE
</td>
</tr>
<tr>
<td style="text-align:left;">
Hypermethylation
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Hypermethylation + UP + UP
</td>
<td style="text-align:left;">
TPDE
</td>
<td style="text-align:left;">
TPDE
</td>
<td style="text-align:left;">
TPDE
</td>
</tr>
<tr>
<td style="text-align:left;">
Hypomethylation
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Hypomethylation + UP + UP
</td>
<td style="text-align:left;">
MDE
</td>
<td style="text-align:left;">
MDE
</td>
<td style="text-align:left;">
MDE
</td>
</tr>
<tr>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
No Change + UP + UP
</td>
<td style="text-align:left;">
TPDE
</td>
<td style="text-align:left;">
TPDE
</td>
<td style="text-align:left;">
TPDE
</td>
</tr>
</tbody>
</table>
</div>
<p><br>
Lastly, it is worth mentioning that in case where only RNAseq and
proteomics data are available and
<strong><em>sircleRCM_RP()</em></strong> was used, all the above stays
the same, yet all the flows and resulting SiRCle clusters that include
DNA-methylation are not available.<br>
Using <strong><em>sircleRCM_2Cond()</em></strong> based on any two input
data, will provide RG1_All, RG2_Significant taking into account genes
that are significant (UP, DOWN, significant positive, significant
negative) and RG3_SignificantChange only takes into account genes that
have significant changes (UP, DOWN).<br><br><br><br></p>
<div class="progress progress-striped active">
<div class="progress-bar" style="width: 100%">

</div>
</div>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<pre><code><span><span class="co">## <span style="color: #00BBBB; font-weight: bold;">─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">##  <span style="color: #555555; font-style: italic;">setting </span> <span style="color: #555555; font-style: italic;">value</span></span></span>
<span><span class="co">##  version  R version 4.4.1 (2024-06-14 ucrt)</span></span>
<span><span class="co">##  os       Windows 10 x64 (build 19045)</span></span>
<span><span class="co">##  system   x86_64, mingw32</span></span>
<span><span class="co">##  ui       RTerm</span></span>
<span><span class="co">##  language en</span></span>
<span><span class="co">##  collate  English_United Kingdom.utf8</span></span>
<span><span class="co">##  ctype    English_United Kingdom.utf8</span></span>
<span><span class="co">##  tz       Europe/Berlin</span></span>
<span><span class="co">##  date     2025-01-08</span></span>
<span><span class="co">##  pandoc   3.2 @ C:/Program Files/RStudio/resources/app/bin/quarto/bin/tools/ (via rmarkdown)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## <span style="color: #00BBBB; font-weight: bold;">─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">##  <span style="color: #555555; font-style: italic;">package    </span> <span style="color: #555555; font-style: italic;">*</span> <span style="color: #555555; font-style: italic;">version</span> <span style="color: #555555; font-style: italic;">date (UTC)</span> <span style="color: #555555; font-style: italic;">lib</span> <span style="color: #555555; font-style: italic;">source</span></span></span>
<span><span class="co">##  bslib         0.8.0   <span style="color: #555555;">2024-07-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  cachem        1.1.0   <span style="color: #555555;">2024-05-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  cli           3.6.3   <span style="color: #555555;">2024-06-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  colorspace    2.1-1   <span style="color: #555555;">2024-07-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  desc          1.4.3   <span style="color: #555555;">2023-12-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  digest        0.6.37  <span style="color: #555555;">2024-08-19</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  dplyr       * 1.1.4   <span style="color: #555555;">2023-11-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  evaluate      1.0.1   <span style="color: #555555;">2024-10-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.2)</span></span></span>
<span><span class="co">##  fastmap       1.2.0   <span style="color: #555555;">2024-05-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  forcats     * 1.0.0   <span style="color: #555555;">2023-01-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  fs            1.6.4   <span style="color: #555555;">2024-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  generics      0.1.3   <span style="color: #555555;">2022-07-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  ggplot2     * 3.5.1   <span style="color: #555555;">2024-04-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  glue          1.7.0   <span style="color: #555555;">2024-01-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  gtable        0.3.6   <span style="color: #555555;">2024-10-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.2)</span></span></span>
<span><span class="co">##  hms           1.1.3   <span style="color: #555555;">2023-03-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  htmltools     0.5.8.1 <span style="color: #555555;">2024-04-04</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  htmlwidgets   1.6.4   <span style="color: #555555;">2023-12-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  jquerylib     0.1.4   <span style="color: #555555;">2021-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  jsonlite      1.8.9   <span style="color: #555555;">2024-09-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  kableExtra  * 1.4.0   <span style="color: #555555;">2024-01-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  knitr         1.49    <span style="color: #555555;">2024-11-08</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.2)</span></span></span>
<span><span class="co">##  lifecycle     1.0.4   <span style="color: #555555;">2023-11-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  lubridate   * 1.9.3   <span style="color: #555555;">2023-09-27</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  magrittr      2.0.3   <span style="color: #555555;">2022-03-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  munsell       0.5.1   <span style="color: #555555;">2024-04-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  pillar        1.10.1  <span style="color: #555555;">2025-01-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  pkgconfig     2.0.3   <span style="color: #555555;">2019-09-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  pkgdown       2.1.1   <span style="color: #555555;">2024-09-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  purrr       * 1.0.2   <span style="color: #555555;">2023-08-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  R6            2.5.1   <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  ragg          1.3.3   <span style="color: #555555;">2024-09-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  readr       * 2.1.5   <span style="color: #555555;">2024-01-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  rlang         1.1.4   <span style="color: #555555;">2024-06-04</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  rmarkdown   * 2.29    <span style="color: #555555;">2024-11-04</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.2)</span></span></span>
<span><span class="co">##  rstudioapi    0.17.1  <span style="color: #555555;">2024-10-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.2)</span></span></span>
<span><span class="co">##  sass          0.4.9   <span style="color: #555555;">2024-03-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  scales        1.3.0   <span style="color: #555555;">2023-11-28</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  sessioninfo   1.2.2   <span style="color: #555555;">2021-12-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  stringi       1.8.4   <span style="color: #555555;">2024-05-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.0)</span></span></span>
<span><span class="co">##  stringr     * 1.5.1   <span style="color: #555555;">2023-11-14</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  svglite       2.1.3   <span style="color: #555555;">2023-12-08</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  systemfonts   1.1.0   <span style="color: #555555;">2024-05-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  textshaping   0.4.0   <span style="color: #555555;">2024-05-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  tibble      * 3.2.1   <span style="color: #555555;">2023-03-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  tidyr       * 1.3.1   <span style="color: #555555;">2024-01-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  tidyselect    1.2.1   <span style="color: #555555;">2024-03-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  tidyverse   * 2.0.0   <span style="color: #555555;">2023-02-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  timechange    0.3.0   <span style="color: #555555;">2024-01-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  tzdb          0.4.0   <span style="color: #555555;">2023-05-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  vctrs         0.6.5   <span style="color: #555555;">2023-12-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  viridisLite   0.4.2   <span style="color: #555555;">2023-05-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  withr         3.0.2   <span style="color: #555555;">2024-10-28</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.2)</span></span></span>
<span><span class="co">##  xfun          0.49    <span style="color: #555555;">2024-10-31</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.2)</span></span></span>
<span><span class="co">##  xml2          1.3.6   <span style="color: #555555;">2023-12-04</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">##  yaml          2.3.10  <span style="color: #555555;">2024-07-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">## </span></span>
<span><span class="co">## <span style="color: #555555;"> [1] C:/Users/chris/AppData/Local/R/win-library/4.4</span></span></span>
<span><span class="co">## <span style="color: #555555;"> [2] C:/Program Files/R/R-4.4.1/library</span></span></span>
<span><span class="co">## </span></span>
<span><span class="co">## <span style="color: #00BBBB; font-weight: bold;">──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span></code></pre>
</div>
<div class="section level2">
<h2 class="unnumbered" id="bibliography">Bibliography<a class="anchor" aria-label="anchor" href="#bibliography"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-Mora_Schmidt2024" class="csl-entry">
Mora, Ariane, Christina Schmidt, Brad Balderson, Christian Frezza, and
Mikael Bodén. 2024. <span>“SiRCle (Signature Regulatory Clustering)
Model Integration Reveals Mechanisms of Phenotype Regulation in Renal
Cancer.”</span> <em>Genome Medicine</em> 16 (1): 144. <a href="https://doi.org/10.1186/s13073-024-01415-3" class="external-link">https://doi.org/10.1186/s13073-024-01415-3</a>.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Ariane Mora, Christina Schmidt, Brad Balderson, Christian Frezza, Mikael Boden.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
